mesonbuild/mparser.py

```diff
@@ -193,10 +193,9 @@ def lex(self, filename: str) -> T.Generator[Token, None, None]:
                     elif tid in {'string', 'fstring'}:
                         # Handle here and not on the regexp to give a better error message.
                         if match_text.find("\n") != -1:
-                            msg = ParseException("Newline character in a string detected, use ''' (three single quotes) "
-                                                 "for multiline strings instead.\n"
-                                                 "This will become a hard error in a future Meson release.",
-                                                 self.getline(line_start), lineno, col)
+                            msg = ("Newline character in a string detected, use ''' (three single quotes) "
```

Comment:
Taking a closer look at this...

The ParseException generates a message string that includes the actual line which triggered the issue, with a caret pointing at the exact position of the issue.

The commit message is wrong to claim that passing the message directly is the same if we don't raise the exception.

To quell mypy, what we could do is `mlog.warning(str(msg), ...)`

Reply:
I've split the code for generating that message out into a helper function and used that here instead.